users/src/api/services/user.service.js

```diff
@@ -0,0 +1,61 @@
+const boom = require('@hapi/boom');
+const bcrypt = require('bcrypt');
+
+const User = require('../models/user.model');
+
+class UserService {
+
+    constructor() {}
+
+    async create(user) {
+        
+        if(await this.findByEmail(user.email)) {
+            throw boom.conflict('Email already exists');
+        }
+
+        const hash = await bcrypt.hash(user.password, 10);
+        const newUser = new User({
+            ...user,
+            password: hash
+        });
+    
+        await newUser.save();
+        user = newUser.toObject();
+        delete user.password;
+        return user;
+    }
+
+    async find() {
+        const userList = await User.find();
+        return userList;
+    }
+
+    async findByEmail(email) {
+        let user = await User.findOne({'email':`${email}`});
+        return user;
+    }
+
+    async findOne(id) {
+        let user = await User.findById(id);
+        if(!user) {
+            throw boom.notFound('User not found');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ce-box/CE5508-astraoke-api/pullRequest?prid=8478842)
```suggestion
            throw boom.notFound("User not found");
```